### DISCUSSION TITLE
k8s log source questions

### DISCUSSION THREAD

**Discussion by atibdialpad** [vectordotdev/vector#7480] [Q&A] [answered]
@jszwedko  I had couple of questions around the usage of vector's k8s source :
1. Do I need to mount my container logs to a shared volume (/var/logs/pod_logs) on the kube-node to enable Vector to consume container logs ? 
2. Datadog Agent daemon set allows to add additional tags to log events via setting  environment variable in the daemon set declaration, examples  a) common tags from env [here](https://docs.datadoghq.com/agent/kubernetes/?tab=daemonset#environment-variables), b) append pod labels as tags to log events [here](https://docs.datadoghq.com/agent/kubernetes/?tab=daemonset#tagging) .Is something like that possible to do in Vector ?

**Comment by atibdialpad** [comment, marked as answer]
[Update]
Got the answer for the first question. 
It seems Vector is using [node-level logging technique ](https://kubernetes.io/docs/concepts/cluster-administration/logging/#using-a-node-logging-agent) which requires all pod/container logs to be in a directory. (/var/log/pods or /var/log/containers which is just a symlink to the pod dir) on the node and I see kubelet already creates these directories.

**Comment by atibdialpad** [comment]
[Update 2] On the second question : 
Vector already adds [pod_labels](https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#pod_labels) to the log events like what ddagent does [pod labels as tags](https://docs.datadoghq.com/agent/kubernetes/tag/?tab=containerizedagent#pod-labels-as-tags) 
It would be great if we could add custom tags as well like [DD_TAGS](https://docs.datadoghq.com/agent/kubernetes/?tab=daemonset#environment-variables) and [node labels as tags](https://docs.datadoghq.com/agent/kubernetes/tag/?tab=containerizedagent#node-labels-as-tags)

**Reply by jszwedko** [reply]
Thanks for updating!

I agree that we should allow node labels as tags; I opened https://github.com/timberio/vector/issues/7499

For `DD_TAGS` you can do this now via `remap` with something like:

```toml
[transforms.remap]
type = "remap"
inputs = [ ... ]
source = """
.some_tag = .some_value
"""
```

Would that meet your needs? One notable difference from the Datadog agent is that we only support key/value pairs for tags.
